### PR TITLE
Process historical BAM data

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**

- [x] Functionality to process historical AirQo BAM data.

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [PLAT-1189](https://airqoteam.atlassian.net/browse/PLAT-1189)
    
**_HOW DO I TEST OUT THIS PR?_**

1. Setup your virtual environment and install all the dependencies in the `requirements.txt` file
2. Download the historical data from `gs://airqo-bam-data/bam_data.csv` and save it as `airqo_historical_bam_data.csv` in the airflow directory.
3. Run the command below to process and save the processed data in `airqo_bam_bigquery_data.csv`

```python
python main.py --action=airqo_historical_bam_data
```
**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
N/A

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A

### CODE REVIEW

**Review by Baalmart — COMMENTED** [review]
Thanks @NoahNsimbe  for this! Brilliant stuff! Just left a few comments for your review.

File: src/airflow/main.py [hunk 1]
```diff
@@ -642,6 +649,69 @@ def airqo_mobile_device_measurements():
...
+    data["timestamp"] = data["timestamp"].apply(pd.to_datetime)
+    data = DataValidationUtils.get_valid_values(data)
+    data = AirQoDataUtils.process_bam_data(
+        data=data, data_type=BamDataType.MEASUREMENTS
+    )
+
+    def update_device_details(device_number):
+        device_id = None
+        device = []
+        if device_number == -24517:
```
[the remark this answers is not in the document]
Comment (NoahNsimbe) [reply src/airflow/main.py:690]: Here am not assigning, am comparing values. You can study the content the the file `gs://airqo-bam-data/bam_data.csv`. 
  -24517 maps to the Nakawa BAM
  -24516 maps to the MUK BAM

@Baalmart

[the remark this answers is not in the document]
Comment (NoahNsimbe) [reply src/airflow/main.py:702]: Hi @Baalmart refer to the docstrings for more information on the purpose of the function

[the remark this answers is not in the document]
Comment (Baalmart) [reply src/airflow/main.py:702]: Alright, my bad! Thanks @NoahNsimbe

**Review by Baalmart — APPROVED** [review]
All LGTM. Thanks @NoahNsimbe

**Pull request merged**